packable/packable/src/packable/prefix/btreeset.rs

```diff
@@ -118,15 +121,25 @@ where
 
         // The length of any dynamically-sized sequence must be prefixed.
         let len = B::unpack::<_, VERIFY>(unpacker, &())
-            .map_packable_err(Self::UnpackError::Prefix)?
+            .map_packable_err(UnpackSetError::Prefix)
+            .map_packable_err(Self::UnpackError::from)?
             .into();
 
         let mut set = BTreeSet::new();
 
         for _ in B::Bounds::default()..len {
-            let item = T::unpack::<_, VERIFY>(unpacker, visitor).map_packable_err(Self::UnpackError::Item)?;
+            let item = T::unpack::<_, VERIFY>(unpacker, visitor)
+                .map_packable_err(UnpackSetError::Item)
+                .map_packable_err(Self::UnpackError::from)?;
+            if let Some(last) = set.last() {
+                if last > &item {
```

Comment:
Doesn't that actually makes the contains useless?

Reply:
Hmm, yeah, I think you're right. I think the checks have to be flipped to be useful.

Reply:
You may not like what I have to say but tbh now that we have the Unordered error, I think the contains check is useless and should be removed

Reply:
It's not though, they are two separate errors.

Reply:
Personally if I got an unordered error when the actual problem was that I had a duplicate it would make it harder to debug.